pkg/controller/job/job_controller.go

```diff
@@ -782,17 +783,17 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 		if uncounted == nil {
 			// Legacy behavior: pretend all active pods were successfully removed.
 			deleted = active
-		} else if deleted != active {
+		} else if deleted != active || !satisfiedExpectations {
```

Comment:
I don't have context to review this line... if we skip this because we haven't satisfied expectations yet, are we relying on a retry and eventually satisfying those expectations, or is this a one-time short-circuit

Reply:
We are not relying on retry per-se. We are relying on the pod creation events to add the job to the workqueue again. This is the same thing we do when starting a job to avoid creating extra pods.